### PR TITLE
[SPARK-18852][SS]StreamingQuery.lastProgress should be null when recentProgress is empty

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Right now `StreamingQuery.lastProgress` throws NoSuchElementException and it's hard to be used in Python since Python user will just see Py4jError.

This PR just makes it return null instead.

## How was this patch tested?

`test("lastProgress should be null when recentProgress is empty")`

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala [hunk 1]
```diff
@@ -312,6 +313,24 @@ class StreamingQuerySuite extends StreamTest with BeforeAndAfter with Logging {
     )
   }
 
+  test("lastProgress should be null when recentProgress is empty") {
+    BlockingSource.latch = new CountDownLatch(1)
```
Comment (brkyvz) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala]: Glad that you got to re-use `BlockingSource`!

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala [hunk 2]
```diff
@@ -312,6 +313,24 @@ class StreamingQuerySuite extends StreamTest with BeforeAndAfter with Logging {
     )
   }
 
+  test("lastProgress should be null when recentProgress is empty") {
```
Comment (brkyvz) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala]: should we add a similar test to python just to make sure it returns `None`? Do we need to update any docs?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala [hunk 3]
```diff
@@ -102,7 +102,7 @@ trait ProgressReporter extends Logging {
 
   /** Returns the most recent query progress update. */
```
Comment (brkyvz) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala]: Maybe add `returns null if there were no progress updates`

**Comment by brkyvz** [comment]
Left one comment on updating docs, otherwise LGTM

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by brkyvz** [comment]
LGTM!

**Pull request closed without merging**